# bps_fbi_sp_ecoli


<!-- WARNING: THIS FILE WAS AUTOGENERATED! DO NOT EDIT! -->

[![PyPI
version](https://badge.fury.io/py/bps-fbi-sp-ecoli.svg)](https://badge.fury.io/py/bps-fbi-sp-ecoli)
[![License:
MIT](https://img.shields.io/badge/License-MIT-yellow.svg)](https://opensource.org/licenses/MIT)
[![Anaconda-Server
Badge](https://anaconda.org/kalilamali/bps_fbi_sp_ecoli/badges/version.svg)](https://anaconda.org/kalilamali/bps_fbi_sp_ecoli)

## Installation

Note that this program requires KMA, which is not available on pip. you
would need to install it via [conda](https://anaconda.org/bioconda/kma)
or [github](https://bitbucket.org/genomicepidemiology/kma/src/master/).

### Installation with pip

``` sh
conda create /.venv python=3.11
pip install bps_fbi_sp_ecoli
conda install bioconda::kma
```

### Installation with conda

``` sh
conda create /.venv python=3.11
conda install kalilamali::bps_fbi_sp_ecoli
pip install requirements.txt
```

## How to use

Determines the serotype and virulence in ecoli through kmer alignment

``` python
! FBI_run_bifrostecolipostkma -h
```

``` python
#! FBI_run_bifrostecolipostkma --input input/231116_CUSTOM_SSIFBI_ECOLI --sample_sheet sample_sheet.xlsx --output output --db_path input/db/ecoligenes --execute
```

``` python
! FBI_run_qcecolisummary -h
```

``` python
#! FBI_run_qcecolisummary --input output/231116_CUSTOM_SSIFBI_ECOLI --output output --execute
```
